db/model/Variable.ts

```diff
@@ -280,23 +285,110 @@ export async function updateAllChartsThatInheritFromIndicator(
                     cc.full = ?,
                     cc.updatedAt = ?,
                     c.updatedAt = ?
-                WHERE c.id = ?
+                WHERE cc.id = ?
             `,
-            [JSON.stringify(fullConfig), updatedAt, updatedAt, chart.chartId]
+            [
+                JSON.stringify(fullConfig),
+                updatedAt,
+                updatedAt,
+                chart.chartConfigId,
+            ]
         )
     }
 
     // let the caller know if any charts were updated
     return inheritingCharts
 }
 
+async function findAllMultiDimViewsThatInheritFromIndicator(
+    trx: db.KnexReadonlyTransaction,
+    variableId: number
+): Promise<
+    {
+        chartConfigId: string
+        patchConfig: GrapherInterface
+        isPublished: boolean
+    }[]
+> {
+    const charts = await db.knexRaw<{
+        chartConfigId: DbPlainMultiDimXChartConfig["chartConfigId"]
+        patchConfig: DbRawChartConfig["patch"]
+        isPublished: boolean
+    }>(
+        trx,
+        `-- sql
+            SELECT
+                mdxcc.chartConfigId as chartConfigId,
+                cc.patch as patchConfig,
+                md.published as isPublished
+            FROM multi_dim_data_pages md
+                JOIN multi_dim_x_chart_configs mdxcc ON mdxcc.multiDimId = md.id
+                JOIN chart_configs cc ON cc.id = mdxcc.chartConfigId
+            WHERE mdxcc.variableId = ?
+        `,
+        [variableId]
+    )
+    return charts.map((chart) => ({
+        ...chart,
+        patchConfig: parseChartConfig(chart.patchConfig),
+    }))
+}
+
+export async function updateAllMultiDimViewsThatInheritFromIndicator(
+    trx: db.KnexReadWriteTransaction,
+    variableId: number,
+    {
+        updatedAt,
+        patchConfigETL,
+        patchConfigAdmin,
+    }: {
+        updatedAt: Date
+        patchConfigETL?: GrapherInterface
+        patchConfigAdmin?: GrapherInterface
+    }
+): Promise<
+    {
+        chartConfigId: string
+        patchConfig: GrapherInterface
+        isPublished: boolean
+    }[]
+> {
```

Comment:
This inline type declaration is a bit unwieldy - maybe pull out the things you need into standalone types and/or define them as partials from other types?

Reply:
Ah, I know, but there are many different inline types in this file and giving them names would be even worse, IMO.